tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Smoke tests for NVM Express Controller.
+#
+# - Install nvme-cli which is a tool to manage NVM Express
+# - Check sys path that should be hold details
+# - Check if we have any NVMe controllers on the PCI bus
+# - Check if we have NVMe devices
+# - Check if we have namespace
+# - Send Identify Admin Command to the NVMe controller
+# - Issue a Read command both using the helper and manually
+# - Compare it to what we have with a regular dd
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "installbasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use scheduler 'get_test_suite_data';
+
+sub _smoke_checks {
+    my $nvm_test_data = shift;
+    assert_script_run("lspci -nn | grep -i nvm");
+    assert_script_run("test -d /sys/class/nvme");
+    assert_script_run("test -c /dev/nvme || echo 'nvm character devices not found'");
+
+    my @nvm_partitions = split(/\n/, script_output("ls -l /dev/$nvm_test_data->{nvm_disk}*"));
+    foreach (@nvm_partitions) {
+        assert_script_run("test -b $_ || echo 'nvm is not a block device'");
+    }
+}
+
+sub _exercise_nvme_commands_and_validate_output {
+    my $nvm_test_data = shift;
```

Comment:
Again as a private function i can see any problem with the name. i think it is descriptive and clear

Reply:
so the reader will assume that all the private method in this file that don't have _validate_output, do not validate output? That is my logic but you can keep it as it is.

Reply:
I want to add this here to understand my logic. I have decided to break the testing in logical subroutines from now on and extract them in functions with descriptive names. the outcome i want to achieve is that you do not have long sequence of lines in `run` and make it easy to understand while you just read through it. Secondly, it will make more easy in the future for refactoring and avoid maybe duplication. For now everything that make sense and belong to the module is private. but if we find that this piece of code can be reused somewhere else, we can easily extract it and make it public in some library or somewhere else.

Reply:
the fact that is private doesn't mean that it couldn't have a good name which would not distract the reader if some of your private methods validate output or not regardless where this method will be in the near future. Your last answer does not address my point (or I cannot see the connection). is it clear why is confusing or should I rephrase it? Anyway you don't need to change if you don't like, just wondering if my feedback is clear to you.